src/categories.h

```diff
@@ -53,19 +54,43 @@ class CategoryFactory
   };
 
 public:
+  struct NexusCategory
+  {
+    NexusCategory(const QString& name, const int nexusID) : m_Name(name), m_ID(nexusID)
+    {}
+    QString m_Name;
+    int m_ID;
+    int m_CategoryID = -1;
+
+    friend bool operator==(const NexusCategory& LHS, const NexusCategory& RHS)
+    {
+      return LHS.m_ID == RHS.m_ID;
+    }
+
+    friend bool operator==(const NexusCategory& LHS, const int RHS)
+    {
+      return LHS.m_ID == RHS;
+    }
+
+    friend bool operator<(const NexusCategory& LHS, const NexusCategory& RHS)
+    {
+      return LHS.m_ID < RHS.m_ID;
+    }
+  };
+
   struct Category
   {
-    Category(int sortValue, int id, const QString& name,
-             const std::vector<int>& nexusIDs, int parentID)
+    Category(int sortValue, int id, const QString& name, int parentID,
+             std::vector<NexusCategory> nexusCats)
         : m_SortValue(sortValue), m_ID(id), m_Name(name), m_HasChildren(false),
-          m_NexusIDs(nexusIDs), m_ParentID(parentID)
+          m_ParentID(parentID), m_NexusCats(nexusCats)
```

Comment:
`nexusCats` should be moved here.

Reply:
I'm assuming you are wanting to simply replace the old 'nexusIDs' but I moved this because it matched the order of the data in the CategoryDialog table.

Therefore I'm not sure I agree with simply replacing the old parameter, as this did have a reason.

Reply:
No I was simply saying that this should be `m_NexusCats(std::move(nexusCats))` since you are taking `nexusCats` by value, but that's a very minor thing. ;)